certora/funnels/automation/ConduitMover.spec

```diff
@@ -0,0 +1,379 @@
+    numOtherBefore, hopOtherBefore, zzzOtherBefore, lotOtherBefore = configs(otherAddr, otherAddr_2, otherAddr_3);
+
+    bytes32 withdrawIlkBefore = withdrawIlk;
+    address withdrawGemBefore = withdrawGem;
+    mathint withdrawAmountBefore = withdrawAmount;
+    bytes32 depositIlkBefore = depositIlk;
+    address depositGemBefore = depositGem;
+    mathint depositAmountBefore = depositAmount;
+
+    move(e, from, to, gem);
+
+    mathint wardsAfter = wards(anyAddr);
+    mathint budsAfter = buds(anyAddr);
+    mathint numFromToGemAfter; mathint hopFromToGemAfter; mathint zzzFromToGemAfter; mathint lotFromToGemAfter;
+    numFromToGemAfter, hopFromToGemAfter, zzzFromToGemAfter, lotFromToGemAfter = configs(from, to, gem);
+    mathint numOtherAfter; mathint hopOtherAfter; mathint zzzOtherAfter; mathint lotOtherAfter;
+    numOtherAfter, hopOtherAfter, zzzOtherAfter, lotOtherAfter = configs(otherAddr, otherAddr_2, otherAddr_3);
+
+    assert wardsAfter == wardsBefore, "move did not keep unchanged every wards[x]";
+    assert budsAfter == budsBefore, "move did not keep unchanged every buds[x]";
+    assert numFromToGemAfter == numFromToGemBefore - 1, "move did not decrease configs[from][to][gem].num by 1";
+    assert hopFromToGemAfter == hopFromToGemBefore, "move did not keep unchanged configs[from][to][gem].hop";
+    assert zzzFromToGemAfter == to_mathint(e.block.timestamp), "move did not set configs[from][to][gem].zzz to block.timestamp";
+    assert lotFromToGemAfter == lotFromToGemBefore, "move did not keep unchanged configs[from][to][gem].lot";
+    assert numOtherAfter == numOtherBefore, "move did not keep unchanged the rest of configs[x][y][z].num";
+    assert hopOtherAfter == hopOtherBefore, "move did not keep unchanged the rest of configs[x][y][z].hop";
+    assert zzzOtherAfter == zzzOtherBefore, "move did not keep unchanged the rest of configs[x][y][z].zzz";
+    assert lotOtherAfter == lotOtherBefore, "move did not keep unchanged the rest of configs[x][y][z].lot";
+    assert from != buffer => withdrawAddr == from, "move did not execute the withdraw external call to the correct 'from' contract";
+    assert from != buffer => withdrawIlk == ilk(), "move did not pass the correct ilk to the withdraw external call";
+    assert from != buffer => withdrawGem == gem, "move did not pass the correct gen to the withdraw external call";
+    assert from != buffer => to_mathint(withdrawAmount) == lotFromToGemBefore, "move did not pass the correct amount to the withdraw external call";
+    assert from == buffer => withdrawIlk == withdrawIlkBefore, "move did execute the withdraw external call when it did not correspond";
+    assert from == buffer => withdrawGem == withdrawGemBefore, "move did execute the withdraw external call when it did not correspond 2";
+    assert from == buffer => to_mathint(withdrawAmount) == withdrawAmountBefore, "move did execute the withdraw external call when it did not correspond 3";
```

Comment:
Technically any one of these should be sufficient to catch an inappropriate `withdraw` call; a cleaner version would be to have `withdrawCalled` boolean ghost that is set in the summary and check that, similar to other specs. An even better option is to have a `mathint` ghost, e.g. `withdrawCounter`, that is incremented in the summary and assert that it increases by exactly one if `from != buffer` and is unchanged if `from == buffer` (ensures that in the case an external call to `withdraw` is made, _only_ one call is made and not 2 or more).

Reply:
Counters were added, I prefer to keep the params so we make sure they are passed correctly.